pkgs/development/python-modules/xtensor-python/default.nix

```diff
@@ -10,26 +10,36 @@
 
 buildPythonPackage rec {
   pname = "xtensor-python";
-  version = "0.25.1";
+  version = "0.26.1";
+
+  format = "other"; # Just CMake
 
   src = fetchFromGitHub {
     owner = "xtensor-stack";
     repo = pname;
     rev = version;
-    sha256 = "17la76hn4r1jv67dzz8x2pzl608r0mnvz854407mchlzj6rhsxza";
+    sha256 = "sha256-kLFt5Ah5/ZO6wfTZQviVXeIAVok+/F/XCwpgPSagOMo=";
   };
 
   nativeBuildInputs = [ cmake pybind11 ];
+  checkInputs = [ gtest ]; # find_package fails if we use nativeCheckInputs
+  nativeCheckInputs = [ gtest ];
+
+  cmakeFlags = [
+    "-DBUILD_TESTS=ON"
+  ];
 
   propagatedBuildInputs = [ xtensor numpy ];
 
   dontUseSetuptoolsBuild = true;
   dontUsePipInstall = true;
   dontUseSetuptoolsCheck = true;
 
-  nativeCheckInputs = [
-    gtest
-  ];
+  # NOTE: buildPythonPackage resets doCheck and ignores preCheck
```

Comment:
```suggestion
  # NOTE: buildPythonPackage overwrites installCheckPhase
```

Reply:
But it literally does reset doCheck and ignore preCheck 🤷🏻

Reply:
I can't find that in code https://github.com/NixOS/nixpkgs/blob/master/pkgs/development/interpreters/python/mk-python-derivation.nix#L166

Reply:
I think you are linking the correct line. What I meant by buildPythonPackage overwriting doCheck is that even if I set `doCheck = true`, the final derivation is going to have `doCheck = false` and `preCheck` won't run. The comment refers to me using `preInstallCheck` instead of `preCheck`. I'll rewrite the comment to clarify this